src/state/machines/guards/buy.ts

```diff
@@ -23,13 +23,18 @@ export const BuyMachineGuards = {
           ?.hand.filter(({ uid }) => turn.playedCards.includes(uid))
           .filter(({ type }) => type === "plant") as PlantCard[]) ?? [];
 
-      return playedPlants.filter(({ biomes }) => intersection(biomes, requiredBiomes).length > 0).length >= 2;
+      return playedPlants.filter(({ biomes }) => requiredBiomes.every((b) => biomes.includes(b))).length >= 2;
     }
 
     return false;
   },
 
-  canBorrow: ({ context: { turn } }: { context: GameState }) => turn.borrowedCount < turn.borrowedLimit,
+  belowBorrowLimit: ({ context: { turn } }: { context: GameState }) => turn.borrowedCount < turn.borrowedLimit,
+  playerHasElement: ({ context }: { context: GameState }, elName: ElementCard["name"]) => {
```

Comment:
This does not take into account that a player can still borrow an element of the same type of one they have in their hand if the card in their hand is exhausted. I believe this is the rule but it might be something we have to clear up with Albert